src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/AbstractHostObject.cs

```diff
@@ -7,12 +7,22 @@
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
 {
+
     /// <summary>
     /// Base type for language service host object for cross targeting projects.
     /// </summary>
-    internal abstract class AbstractHostObject : IVsHierarchy, IVsContainedLanguageProjectNameProvider
+    internal abstract class AbstractHostObject : IVsHierarchy, IVsContainedLanguageProjectNameProvider, IVsProject
     {
-        protected abstract IVsHierarchy InnerHierarchy { get; }
+        protected AbstractHostObject(object innerHostObject)
```

Comment:
The only problem with that approach is that we will basically end up creating a parameter for each interface that this type implements. Unfortunately, the host object exposed by CPS or the Hierarchy exposed by Roslyn's workspace context is strongly typed as IVsHierarchy, though the consumers know the implementing host object type implements these additional interfaces and they just do explicit casts (see [here](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/Venus/ContainedDocument.cs,98) for an example). Unless we expose a VS specific host object type through CPS and Roslyn project system, we will always end up with this opaque host object with no strong type.